lib/board.rb

```diff
@@ -31,4 +31,55 @@ def update_board(player_selection, computer_selection)
     @grid[player_selection][@grid[player_selection].rindex('.')] = 'X'
     @grid[computer_selection][@grid[computer_selection].rindex('.')] = 'O'
   end
+
+  def check_for_end_game
+    vertical = @grid.values
+    horizontal = @grid.values.transpose
+    diagonal = create_diagonals
+    result = :keep_playing
+    result = :win if check_for_win(vertical, horizontal, diagonal) == true
+    result = :loss if check_for_loss(vertical, horizontal, diagonal) == true
+    result = :draw if check_for_draw == true
+    result
+  end
+
+  def check_for_win(vertical, horizontal, diagonal)
+    [vertical, horizontal, diagonal].each do |grid|
+      grid.each do |values|
+        values.each_cons(4) do |quad|
+          return true if quad == ['X', 'X', 'X', 'X']
+        end
+      end
+    end
+  end
+
+  def check_for_loss(vertical, horizontal, diagonal)
+    [vertical, horizontal, diagonal].each do |grid|
+      grid.each do |values|
+        values.each_cons(4) do |quad|
+          return true if quad == ['O', 'O', 'O', 'O']
+        end
+      end
+    end
+  end
+
+  def check_for_draw
+    grid.values.flatten.none?{|element| element == '.'}
+  end
+
+  def create_diagonals
```

Comment:
Consider coming back during refactoring to explore an algorithmic solution to finding diagonals in the grid.